apps/omg_watcher_info/lib/omg_watcher_info/release_tasks/ethereum_tasks/add_eth_height_to_eth_events.ex

```diff
@@ -0,0 +1,123 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherInfo.ReleaseTasks.EthereumTasks.AddEthereumHeightToEthEvents do
+  @moduledoc """
+  Running in May 2020.
+  `eth_height` is currently not persisted in the Watcher Info DB for `eth_events`
+  This module will add `eth_height` to all persisted `eth_events` where this value is non-existent.
+  """
+  use Ecto.Migration
+  alias OMG.Eth.Encoding
+  alias OMG.WatcherInfo.DB
+
+  import Ecto.Query, only: [from: 2]
+
+  @max_db_rows 100
+  @max_eth_requests 25
+
+  def run() do
+    DB.Repo.transaction(fn ->
```

Comment:
Good point. I have this because the [documentation](https://hexdocs.pm/ecto/Ecto.Repo.html#c:stream/2) on streams states "SQL adapters, such as Postgres and MySQL, can only enumerate a stream inside a transaction." 

On the one hand it is a single transaction. On the other it's a stream. Not sure if there is a vulnerability there. 🤔

Reply:
Ah I didn't know that. In that case I think it's fine. This only gets triggered on watcher_info start up and will reach this code only once per environment per its lifetime anyway.